### ISSUE TITLE
Can the firebase document ref path be customized without using the function?

### ISSUE THREAD

**Issue by prasenjitpickcel** [transistorsoft/flutter_background_geolocation_firebase#5]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by christocracy** [comment]
No.

**Comment by basudevs** [comment]
Hi,
We are trying to get the gelocation when the app is at background and push the data to firestore without using any cloud functions. (We are comfortable in using functions is, we had seen issues with cloud functions in large scale, and is an additional failure point for copying the data).From your above answer looks like this is not possible directly. 
Alternatively, We would like to know if we can access location data from your library using broadcast services every 15 minutes. So that we can write our flutter code to post it to firestore.

**Comment by christocracy** [comment]
I think it’s unnecessarily paranoid to write off cloud functions.

https://cloud.google.com/functions/docs/bestpractices/retries

Otherwise, see wiki Android Headless Mode

**Comment by christocracy** [comment]
>Automatic retrying is not available for HTTP functions.

A cloud-function to simply re-save the data with a different schema *doesn't involve HTTP*.  And as a result, these invocations [are *incredibly* inexpensive](https://cloud.google.com/functions/pricing), at about $3.20 / 10M invocations.

**Comment by ajanieniolasolomon** [comment]
@christocracy  how can I pass my usersID along with the geolocation data. Tried doing it like locationsCollection: "/locations/{$usserid}" but I notice it keeps overriding it n the cloud function.pls is there a fix because I want to attach a geolocation record to a userId

**Comment by christocracy** [comment]
@ajanieniolasolomon Your collection-name needs to have an *odd* number of parts, yours has *two* (even).

eg:  `locationsCollection: "/users/{$userid}/locations"`

**Comment by ajanieniolasolomon** [comment]
Yes I did try this with cloud function no content was created on my users
location collection

**Comment by christocracy** [comment]
Are you watching the application logs?
